### PR TITLE
#3922 8.1 dep upgrades 3

### LINKED ISSUE #3911: Nationality/Ethnicity should be saved from patient lookups
## Is your feature request related to a problem? Please describe.

The patient lookup isn't mapping nationality and ethnicities

## Describe the solution you'd like

Add the fields

## Implementation

TBD

## Describe alternatives you've considered

TBD

## Additional context

TBD

### LINKED ISSUE #3922: 8.1 dependency upgrades
## The suggestion

Been awhile since we did a round of dependency upgrades. We have some time now, with no clear deadline for 8.1 to get these done.

**Why?**
So we don't get to far behind.

## Describe alternatives you've considered

Doing nothing

## Additional context

N/A

### PR DESCRIPTION
**Pull request by katherine-sussol**
Fixes #3922

## Change summary

- Main changes needed were for `@react-navigation/core` (action creators were moved to a different package and changed a bit - jumping between screens _seems_ to still work).
- Others bumped with no changes made:
  - Babel (7.14.0)
  - react-native-device-info (8.1.2)
  - redux (4.1.0)
  - react-native-svg (12.1.1)

These are the remaining outstanding, being too slow and updates for some are coming out already after bumping 😹 :
image

## Testing

Steps to reproduce or otherwise test the changes of this PR:

- [ ] Build the app
- [ ] Run the app and try a bit of everything, hopefully, nothing new was broken

### Related areas to think about
- Manage stocktakes doesn't update quantities properly but found that was a separate issue (https://github.com/openmsupply/mobile/issues/4010)
- I don't know what happened to the sync button in the sync modal, will investigate that separately as it exists in `develop`
image

### CODE REVIEW

**Comment by mark-prins** [comment]
@katherine-sussol possibly the issue number isn't #3911 ?

**Comment by mark-prins** [comment]
Still looking - but I can't build this branch. The error I get is:

```> Task :app:checkDebugAarMetadata FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:checkDebugAarMetadata'.
> Could not resolve all files for configuration ':app:debugRuntimeClasspath'.
   > Could not resolve com.github.Polidea:MultiPlatformBleAdapter:0.1.7.
     Required by:
         project :app > project :react-native-ble-plx
      > Could not resolve com.github.Polidea:MultiPlatformBleAdapter:0.1.7.
         > Could not get resource 'https://www.jitpack.io/com/github/Polidea/MultiPlatformBleAdapter/0.1.7/MultiPlatformBleAdapter-0.1.7.pom'.
            > Could not GET 'https://www.jitpack.io/com/github/Polidea/MultiPlatformBleAdapter/0.1.7/MultiPlatformBleAdapter-0.1.7.pom'.
               > Read timed out
```

**Comment by katherine-sussol** [comment]
> @katherine-sussol possibly the issue number isn't #3911 ?

Oops, thanks - I'm not sure where #3911 came from!

**Review by joshxg — COMMENTED** [review]
I'm also still looking but getting a dex thing:

```
com.android.builder.dexing.DexArchiveMergerException: Error while merging dex archives: 
Learn how to resolve the issue at https://developer.android.com/studio/build/dependencies#duplicate_classes.
Type androidx.activity.R$attr is defined multiple times: /Users/joshuagriffin/repos/mobile/android/app/build/intermediates/project_dex_archive/debug/out/androidx/activity/R$attr.dex, /Users/joshuagriffin/repos/mobile/android/app/build/intermediates/project_dex_archive/debug/out/0d9d266302c19bee704641403698dd92012d54bddc3308320492e3c3a7dee06a_1.jar:classes.dex
```

**Comment by katherine-sussol** [comment]
I've gotten the dex thing when I forgot to run `cd android && ./gradlew clean` when swapping from a branch on an older RN version, but I've never seen the `react-native-ble-plx` one 🤔  hmm

File: src/navigation/actions.js [hunk 1]
```diff
@@ -266,26 +266,27 @@ export const gotoStocktakes = () =>
...
+ * @param {String} stocktakeName The name of the stocktake to pass to the next screen.
+ * @param {Object} stocktake The stocktake to pass to the next screen.
  */
 export const gotoStocktakeManagePage = (stocktakeName, stocktake) => dispatch => {
-  const currentRouteName = RootNavigator.getCurrentRouteName();
+  const currentName = RootNavigator.getCurrentRouteName();
 
   const navigationActionCreator =
-    currentRouteName === ROUTES.STOCKTAKES ? NavigationActions.navigate : StackActions.replace;
+    currentName === ROUTES.STOCKTAKES ? CommonActions.navigate : StackActions.replace;
```
Comment (joshxg) [inline src/navigation/actions.js:286]: This ones throwing an error for me and won't let me do a normal stocktake.

Looks like the `StackActions` api has changed also https://reactnavigation.org/docs/stack-actions 

Seems they `CommonActions` able to pass `name, params` or `{ name, params }` but for `StackActions` it is only `name, params` ?

**Review by joshxg — CHANGES_REQUESTED** [review]
Actually my issue turned out to be caching 🙄 

After `react-native clean-mac-cache` and `cd android && ./gradlew clean` I was able to build and everything was OK

File: src/navigation/actions.js (hunk 1 above)
Comment (katherine-sussol) [reply src/navigation/actions.js]: Thanks! That was subtle. Have fixed with 3 commits (24c80c8) was the main one but then I did some copy paste fails so there's also cb3b72f and bfbfe79.

Found some other strange behaviour where deleting multiple unfinalised stocktakes causes a rsod, but I'm not sure if it was this PR yet, will give it a test on develop (and master if it also happens there).

**Comment by katherine-sussol** [comment]
image
Seems to be something new although I don't know where yet. Will have a look...

**Comment by mark-prins** [comment]
Woo! I'm working again. not sure why.. but I think safe to ignore the build error I was getting.

**Comment by katherine-sussol** [comment]
Found the fail, think I did a 'replace all' on the file to rename `routeName` -> `name` and it renamed something I didn't want it to rename 🤣 . Should be good for re-review again ae26403

File: src/navigation/actions.js [hunk 2]
```diff
@@ -266,37 +266,40 @@ export const gotoStocktakes = () =>
...
- * Action creator for navigating to the SupplierRequisition screen.
+ * Action creator for navigating to the Stocktake Manage screen.
  *
- * @param {Object} requisition The requisition to pass to the next screen.
+ * @param {String} stocktakeName The name of the stocktake to pass to the next screen.
+ * @param {Object} stocktake The stocktake to pass to the next screen.
  */
 export const gotoStocktakeManagePage = (stocktakeName, stocktake) => dispatch => {
-  const currentRouteName = RootNavigator.getCurrentRouteName();
+  const currentName = RootNavigator.getCurrentRouteName();
```
Comment (mark-prins) [inline src/navigation/actions.js:283]: I think `currentRouteName` has a little more meaning and would be good to retain

**Review by mark-prins — APPROVED** [review]
haven't found any issues 👍

File: src/navigation/actions.js (hunk 2 above)
Comment (katherine-sussol) [reply src/navigation/actions.js]: I agree, I found it a bit odd `react-navigation` moved from calling it `routeName` to just `name` in their API but good to keep the context where we can 😸  cb91d854

**Review by joshxg — APPROVED** [review]
Whoops sorry for blocking!

**Pull request merged**